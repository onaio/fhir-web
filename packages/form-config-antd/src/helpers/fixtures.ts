/* eslint-disable @typescript-eslint/camelcase */
export const manifestRelease4 = {
  identifier: '1.0.30',
  json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
  appId: 'org.smartregister.foo',
  appVersion: '1.2.18',
  createdAt: '2020-06-09T14:40:04.189+02:00',
  updatedAt: '2020-06-09T14:40:04.189+02:00',
};

export const fixManifestReleases = [
  {
    identifier: '1.0.11',
    json: '{"forms_version":"1.0.11","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.14',
    createdAt: '2020-06-08T15:00:38.413+02:00',
    updatedAt: '2020-06-08T15:00:38.413+02:00',
  },
  {
    identifier: '1.0.12',
    json: '{"forms_version":"1.0.12","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.14',
    createdAt: '2020-06-08T15:03:00.950+02:00',
    updatedAt: '2020-06-08T15:03:00.950+02:00',
  },
  {
    identifier: '1.0.16',
    json: '{"forms_version":"1.0.16","identifiers":["blood_screening.json"}',
    appId: 'org.smartregister.reveal',
    appVersion: '1.2.16',
    createdAt: '2020-06-09T14:40:04.189+02:00',
    updatedAt: '2020-06-09T14:40:04.189+02:00',
  },
];

export const manifestFile1 = {
  createdAt: 'Jun 19, 2020, 12:31:37 PM',
  form_relation: '',
  id: '52',
  identifier: 'test-form-1.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'test form',
  module: '',
  version: '1.0.26',
};

export const manifestFile2 = {
  createdAt: 'Jun 19, 2020, 4:23:22 PM',
  form_relation: '',
  id: '53',
  identifier: 'reveal-test-file.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'test publish',
  module: '',
  version: '1.0.27',
};

export const manifestFile3 = {
  createdAt: 'Jun 19, 2020, 6:23:22 PM',
  form_relation: '',
  id: '54',
  identifier: 'foo.json',
  isDraft: false,
  isJsonValidator: false,
  jursdiction: '',
  label: 'foo',
  module: 'baz',
  version: '1.0.28',
};

export const draftFile1 = {
  ...manifestFile1,
  isDraft: true,
  isJsonValidator: false,
};

export const draftFile2 = {
  ...manifestFile2,
  isDraft: true,
  isJsonValidator: false,
};

export const draftFile3 = {
  ...manifestFile3,
  isDraft: true,
  isJsonValidator: false,
};

export const fixManifestFiles = [manifestFile1, manifestFile2];

export const FixManifestDraftFiles = [
  {
    ...fixManifestFiles[0],
    isDraft: false,
  },
  {
    ...fixManifestFiles[1],
    isDraft: false,
  },
];

export const downloadFile = {
  clientForm: {
    json: JSON.stringify(fixManifestFiles[0]),
  },
};
