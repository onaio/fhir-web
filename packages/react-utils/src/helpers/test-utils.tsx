import { store } from '@opensrp/store';
import { Provider } from 'react-redux';
import React from 'react';
import { RoleContext, UserRole, FhirResources, Permit, IamResources } from '@opensrp/rbac';
import { history } from '@onaio/connected-reducer-registry';
import { Router } from 'react-router';

export const superUserRole = new UserRole([...FhirResources, ...IamResources], Permit.MANAGE);

export interface ContextProviderProps {
  children: JSX.Element;
}

export const ContextProvider = (props: ContextProviderProps) => {
  const { children } = props;
  return (
    <Provider store={store}>
      <Router history={history}>
        <RoleContext.Provider value={superUserRole}>{children}</RoleContext.Provider>
      </Router>
    </Provider>
  );
};
