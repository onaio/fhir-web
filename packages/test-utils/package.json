{
  "name": "@opensrp/test-utils",
  "version": "0.0.0",
  "description": "",
  "main": "dist/index.js",
  "types": "dist/types",
  "files": [
    "dist"
  ],
  "publishConfig": {
    "access": "public"
  },
  "repository": "https://github.com/opensrp/web",
  "scripts": {
    "test": "run -T test $INIT_CWD --verbose",
    "tsc": "run -T tsc",
    "lint": "run -T eslint ./**/*.{js,jsx,ts,tsx}",
    "copy": "run -T copyfiles -u 1 \"./src/**/*.{css,html}\" \"./dist/\"",
    "build": "run tsc && run transpile && run copy",
    "transpile": "run -T babel src -d dist --root-mode upward --extensions .ts,.tsx --ignore '**/*.test.ts,**/*.test.tsx,**/tests,**/__tests__'"
  },
  "jest": {
    "automock": false,
    "setupFiles": [
      "../../setupTests"
    ]
  },
  "bugs": {
    "url": "https://github.com/opensrp/web/issues"
  },
  "dependencies": {
    "@onaio/connected-reducer-registry": "^0.0.3",
    "@opensrp/pkg-config": "^0.0.9",
    "@opensrp/rbac": "workspace:^"
  },
  "peerDependencies": {
    "@opensrp/store": "^0.0.10",
    "@testing-library/react": "^14.1.2",
    "@testing-library/user-event": "^13.5.0",
    "react": "^17.0.0 ",
    "react-redux": "^7.2.1",
    "react-router": "^5.2.1",
    "react-router-dom": "^5.2.1"
  },
  "author": "OpenSRP Engineering",
  "license": "Apache-2.0"
}
