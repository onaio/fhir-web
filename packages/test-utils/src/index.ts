export * from './componentWrappers';
export * from './utils'