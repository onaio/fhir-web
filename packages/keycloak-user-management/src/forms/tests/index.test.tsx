import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import flushPromises from 'flush-promises';
import React from 'react';
import fetch from 'jest-fetch-mock';
import { UserForm } from '..';
import * as fixtures from './fixtures';
import { act } from 'react-dom/test-utils';

describe('src/components/UserForm', () => {
  beforeEach(() => {
    fetch.resetMocks();
  });
  it('renders without crashing', () => {
    fetch.once(JSON.stringify([fixtures.keycloakUser]));
    shallow(<UserForm />);
  });

  it('renders correctly', () => {
    fetch.once(JSON.stringify([fixtures.keycloakUser]));
    // looking for each fields
    const wrapper = mount(<UserForm />);

    // user's first name
    const userInput = wrapper.find('input#firstName');
    expect(userInput).toHaveLength(1);
    expect(toJson(userInput)).toMatchSnapshot('first name input');
    wrapper.unmount();
  });

  it('creates object to send correctly for creating new user', async () => {
    fetch.once(JSON.stringify([]));
    const wrapper = mount(<UserForm />);

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });

    // set users's first name
    const nameInput = wrapper.find('input#firstName');
    nameInput.simulate('change', { target: { name: 'firstName', value: 'Test' } });

    // set users last name
    const lastNameInput = wrapper.find('input#lastName');
    lastNameInput.simulate('change', { target: { name: 'lastName', value: 'One' } });

    // set username
    const usernameInput = wrapper.find('input#username');
    usernameInput.simulate('change', { target: { name: 'username', value: 'TestOne' } });

    // set user email
    const emailInput = wrapper.find('input#email');
    emailInput.simulate('change', { target: { name: 'email', value: 'testone@gmail.com' } });

    wrapper.find('form').simulate('submit');
    await act(async () => {
      wrapper.update();
    });

    await new Promise<unknown>((resolve) => setImmediate(resolve));
    wrapper.update();

    const payload = {
      access: {
        manageGroupMembership: false,
        view: false,
        mapRoles: false,
        impersonate: false,
        manage: false,
      },
      disableableCredentialTypes: [],
      email: 'testone@gmail.com',
      emailVerified: false,
      enabled: true,
      firstName: 'Test',
      id: '',
      lastName: 'One',
      notBefore: 0,
      requiredActions: [],
      totp: false,
      username: 'TestOne',
    };

    expect(fetch.mock.calls[0]).toEqual([
      'https://keycloak-test.smartregister.org/auth/realms//users',
      {
        'Cache-Control': 'no-cache',
        Pragma: 'no-cache',
        body: JSON.stringify(payload),
        headers: {
          accept: 'application/json',
          authorization: 'Bearer null',
          'content-type': 'application/json;charset=UTF-8',
        },
        method: 'POST',
      },
    ]);
    wrapper.unmount();
  });

  it('creates object to send correctly for editing user', async () => {
    fetch.once(JSON.stringify([fixtures.keycloakUser]));
    const props = {
      initialValues: fixtures.keycloakUser,
    };
    const wrapper = mount(<UserForm {...props} />);

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });

    // practitioner's name
    const nameInput = wrapper.find('input#firstName');
    nameInput.simulate('change', { target: { name: 'firstName', value: 'Test2' } });

    expect(toJson(wrapper.find('input#username'))).toMatchSnapshot('readonly username edit');

    wrapper.find('form').simulate('submit');

    await act(async () => {
      wrapper.update();
    });

    await new Promise<unknown>((resolve) => setImmediate(resolve));

    const payload = {
      id: 'cab07278-c77b-4bc7-b154-bcbf01b7d35b',
      createdTimestamp: 1600156317992,
      username: 'opensrp',
      enabled: true,
      totp: false,
      emailVerified: false,
      firstName: 'Test2',
      lastName: 'kenya',
      email: 'test@onatest.com',
      disableableCredentialTypes: [],
      requiredActions: [],
      notBefore: 0,
      access: {
        manageGroupMembership: true,
        view: true,
        mapRoles: true,
        impersonate: false,
        manage: true,
      },
    };

    expect(fetch.mock.calls[0]).toEqual([
      'https://keycloak-test.smartregister.org/auth/realms//users/cab07278-c77b-4bc7-b154-bcbf01b7d35b',
      {
        'Cache-Control': 'no-cache',
        Pragma: 'no-cache',
        body: JSON.stringify(payload),
        headers: {
          accept: 'application/json',
          authorization: 'Bearer null',
          'content-type': 'application/json;charset=UTF-8',
        },
        method: 'PUT',
      },
    ]);
    wrapper.unmount();
  });
  it('user is not created if api is down', async () => {
    fetch.mockReject(() => Promise.reject('API is down'));
    const wrapper = mount(<UserForm />);

    await act(async () => {
      await flushPromises();
      wrapper.update();
    });

    // set users's first name
    const nameInput = wrapper.find('input#firstName');
    nameInput.simulate('change', { target: { name: 'firstName', value: 'Test' } });

    // set users last name
    const lastNameInput = wrapper.find('input#lastName');
    lastNameInput.simulate('change', { target: { name: 'lastName', value: 'One' } });

    // set username
    const usernameInput = wrapper.find('input#username');
    usernameInput.simulate('change', { target: { name: 'username', value: 'TestOne' } });

    // set user email
    const emailInput = wrapper.find('input#email');
    emailInput.simulate('change', { target: { name: 'email', value: 'testone@gmail.com' } });

    wrapper.find('form').simulate('submit');
    await act(async () => {
      wrapper.update();
    });

    await new Promise<unknown>((resolve) => setImmediate(resolve));
    wrapper.update();
    expect(document.getElementsByClassName('ant-notification')).toHaveLength(1);
    wrapper.unmount();
  });
});
