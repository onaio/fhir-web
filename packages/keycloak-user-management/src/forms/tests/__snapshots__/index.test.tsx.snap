// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/UserForm creates object to send correctly for editing user: readonly username edit 1`] = `
<input
  className="form-control"
  id="username"
  name="username"
  onBlur={[Function]}
  onChange={[Function]}
  readOnly={true}
  type="text"
  value="opensrp"
/>
`;

exports[`src/components/UserForm renders correctly: first name input 1`] = `
<input
  className="form-control"
  id="firstName"
  name="firstName"
  onBlur={[Function]}
  onChange={[Function]}
  type="text"
  value=""
/>
`;
